### ISSUE TITLE
Some pages won't load when using a reverse proxy

### ISSUE THREAD

**Issue by MakoSDV** [grafana/grafana#11540]
### What Grafana version are you using?
5.0.4
### What datasource are you using?
InfluxDB
### What OS are you running grafana on?
Ubuntu 16.0.4 LTS
### What did you do?
Setup a proxy using Apache to direct https://server/dash to http://127.0.0.1:3000 and updated the domain and root_url paths in Grafana.ini

Related Apache conf settings:
<VirtualHost *:443>
  ...
  ProxyPreserveHost On
  SSLProxyEngine On
  <Location "/dash">
    ProxyPass http://127.0.0.1:3000
    ProxyPassReverse http://127.0.0.1:3000
  </Location>

Related Grafana.ini settings:
 domain = server
 root_url = https://server/dash

### What was the expected result?
Grafana should function normally using the new root_url

### What happened instead?
Some pages load but others do not.

Working: Login, viewing of existing dashboards, Alerting, Configuration, switching organization, user preferences, Manage Playlists, Sign Out

Not working (Page not found 404 error): Create Dashboard, Create Folder, Import, Manage Dashboards, Manage Snapshots

In other words: /board and /boards paths do not load, but all other paths seem to work

I checked the grafana log, but it doesn't say anything other than the 404.  For example:
t=2018-04-09T17:59:12+0000 lvl=info msg="Request Completed" logger=context userId=1 orgId=2 uname=admin method=GET path=/board/new status=404 remote_addr=<ip> time_ms=5 size=15972 referer=https://<server>/dash/login

Please let me know if you need anymore information.

**Comment by MakoSDV** [comment]
I just tested /grafana and it works.  That's messed up.

Is it setup so that the only allowable subpath is /grafana?

**Comment by torkelo** [comment]
strange, no one else has reported any issues with using other names, it is working in setups we have. Mostly run nginx to reverse proxy though. Not sure what could be wrong with your apache config.

**Comment by MakoSDV** [comment]
I just switched it from /grafana to /graf and that works fine as well.

The issues appears to be when using /dash.  I suspect that there is some special handling for the keyword "dashboard" and the paths /dash/board or /dash/boards triggers it and causes the path to get messed up.

**Comment by daniellee** [comment]
This has come up a few times on the community site: https://community.grafana.com/t/404-error-when-try-to-create-a-folder/5637

Don't think there is any easy fix for it though.

**Comment by chook** [comment]
I also experienced issues with proxying under `/dashboard`.
It seems the `/dashboard/dashboard/*` was redirecting to `/dashboard/*`
